### PR TITLE
added forgot password

### PR DESCRIPTION
**Pull request by Devyani606**

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -44,13 +45,38 @@ def create_user(email) # rubocop:disable Metrics/AbcSize
         render_success(message: 'success')
       end
 
+      def forget_password
+        user = User.find_by(email: params[:email])
+        if user
+          user.reset_password_token = SecureRandom.hex
+          user.save
+          redirect_url = params[:redirect_url]
```
Comment (somnath77) [inline app/controllers/api/v1/users_controller.rb:53]: no need to ask redirect_url from params

File: app/views/user_mailer/forgot_password_email.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<p> Hi,<%=@user.first_name %></p>
+
+<p> We have received your request to reset your password. Please click the link below to complete the reset:  
+</p>
+
+<p><%= link_to "click here", "#{ENV['APP_URL']}password_edit/#{@redirect_url}/#{@user.reset_password_token}" %>
```
Comment (somnath77) [inline app/views/user_mailer/forgot_password_email.html.erb:6]: build correct url,  discuss with onkar

**Review by somnath77 — CHANGES_REQUESTED** [review]

File: app/views/user_mailer/forgot_password_email.html.erb (hunk 2 above)
Comment (Devyani606) [reply app/views/user_mailer/forgot_password_email.html.erb:6]: yes

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (Devyani606) [reply app/controllers/api/v1/users_controller.rb:53]: ok

**Review by somnath77 — APPROVED** [review]

**Pull request merged**